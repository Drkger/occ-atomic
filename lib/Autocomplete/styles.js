import spacing from '../subatomic/spacing';

export default {
    autoComplete: {
        position: 'relative'
    },
    droplist: {
        position: 'absolute',
        zIndex: 2,
        top: `calc(100% - ${spacing.xTiny}px)`,
        left: 0,
        width: '100%'
    }
};
