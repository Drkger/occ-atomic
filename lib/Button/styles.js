import colors from '../subatomic/colors';
import fonts from '../subatomic/fonts';
import icons from '../subatomic/icons';

export default {
    btn: {
        alignItems:'flex-start',
        boxSizing:'border-box',
        display:'inline-block',
        marginBottom:'0',
        textAlign:'center',
        whiteSpace:'nowrap',
        verticalAlign:'middle',
        touchAction:'manipulation',
        cursor:'pointer',
        userSelect:'none',
        padding:'0 30px',
        height:'30px',
        lineHeight:'24px',
        fontFamily:fonts.body,
        fontSize:'14px',
        fontWeight:'600',
        position:'relative',
        transition:'0.3s all',
        maxWidth:'100%',
        borderRadius:'5px',
        textDecoration:'none',
        '&:focus': {
            outline:'none'
        },
        '&:hover': {
            textDecoration:'none'
        }
    },
    cont: {
        display:'flex',
        width:'100%',
        height:'100%',
        alignItems:'center',
        justifyContent:'center',
        pointerEvents:'none'
    },
    loadCont: {
        display:'flex',
        width:'100%',
        height:'100%',
        alignItems:'center',
        justifyContent:'center',
        position:'absolute',
        top:'0',
        left:'0',
        pointerEvents:'none'
    },
    // Themes
    main: {
        background:colors.pink,
        color:colors.white,
        border:`1px solid ${colors.pink}`,
        '&:hover': {
            background:colors.pinkH,
            color:colors.white,
            borderColor:colors.pinkH
        },
        '&:active': {
            background:colors.pinkA,
            color:colors.white,
            borderColor:colors.pinkA
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&$loading': {
            background:colors.pinkA,
            borderColor:colors.pinkA
        },
        '&$disabled': {
            background:`${colors.grey1} !important`,
            color:`${colors.white} !important`,
            border:`1px solid ${colors.grey1} !important`
        }
    },
    secondary: {
        background:'transparent',
        color:colors.pink,
        border:`1px solid ${colors.pink}`,
        '&:hover': {
            background:colors.pinkH,
            color:colors.white,
            borderColor:colors.pinkH,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '&:active': {
            background:colors.pinkA,
            color:colors.white,
            borderColor:colors.pinkA,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.pink])) : ''
        },
        '&$loading': {
            background:colors.pinkA,
            borderColor:colors.pinkA
        },
        '&$disabled': {
            background:`${colors.grey1} !important`,
            color:`${colors.white} !important`,
            border:`1px solid ${colors.grey1} !important`
        }
    },
    white: {
        background:'transparent',
        color:colors.white,
        border:`1px solid ${colors.white}`,
        '&:hover': {
            background:'rgba(255,255,255,0.2)',
            color:colors.white,
            borderColor:colors.white,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '&:active': {
            background:'rgba(255,255,255,0.2)',
            color:colors.white,
            borderColor:colors.white,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&$loading': {
            background:'rgba(255,255,255,0.2)',
            borderColor:colors.white
        },
        '&$disabled': {
            background:`transparent !important`,
            color:`rgba(255,255,255,0.6) !important`,
            border:`1px solid rgba(255,255,255,0.6) !important`,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon(['rgba(255,255,255,0.6)'])) : ''
            }
        }
    },
    flat: {
        background:'transparent',
        color:colors.pink,
        border:`1px solid transparent`,
        '&:hover': {
            background:colors.grey3,
            color:colors.pink,
            borderColor:colors.grey3,
        },
        '&:active': {
            background:colors.grey5,
            color:colors.pink,
            borderColor:colors.grey5,
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.pink])) : ''
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.pink]))
        },
        '&$loading': {
            background:colors.white,
            borderColor:colors.white
        },
        '&$disabled': {
            background:'transparent !important',
            color:`${colors.grey1} !important`,
            border:`1px solid transparent !important`,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.grey1])) : ''
            },
            '& $loadIcon': {
                background: icons.base(icons.loading.icon([colors.grey1]))
            },
        }
    },
    blue: {
        background:colors.blue,
        color:colors.white,
        border:`1px solid ${colors.blue}`,
        '&:hover': {
            background:colors.blueH,
            color:colors.white,
            borderColor:colors.blueH,
        },
        '&:active': {
            background:colors.blueA,
            color:colors.white,
            borderColor:colors.blueA,
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&$loading': {
            background:colors.blueA,
            borderColor:colors.blueA
        },
        '&$disabled': {
            background:`${colors.grey1} !important`,
            color:`${colors.white} !important`,
            border:`1px solid ${colors.grey1} !important`
        }
    },
    loading: {
        pointerEvents:'none',
        '& $cont': {
            visibility:'hidden'
        }
    },
    disabled: {
        background:colors.grey1,
        color:colors.white,
        border:`1px solid ${colors.grey1}`,
        cursor:'default',
        pointerEvents:'none',
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&:hover, &:active, &:focus': {
            background:colors.grey1,
            color:colors.white,
            border:`1px solid ${colors.grey1}`
        }
    },
    // Sizes
    md: {
        fontSize:'16px',
        height:'40px',
        lineHeight:'28px',
        '& i': {
            width:'28px',
            height:'28px'
        }
    },
    lg: {
        fontSize:'18px',
        height:'46px',
        lineHeight:'32px',
        '& i': {
            width:'32px',
            height:'32px'
        }
    },
    // Block
    block: {
        display:'block',
        width:'100%'
    },
    // Icon
    icon: {
        width:'24px',
        height:'24px',
        display:'inline-block',
        transition:'0.3s all'
    },
    '@keyframes rotate': {
        from: {transform: 'rotate(0deg)'},
        to: {transform: 'rotate(360deg)'}
    },
    loadIcon: {
        width:'24px',
        height:'24px',
        display:'inline-block',
        transition:'0.3s all',
        animation:'rotate 1s infinite linear',
        background: icons.base(icons.loading.icon())
    }
};