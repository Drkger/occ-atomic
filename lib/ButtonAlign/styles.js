export default {
    sm: {
        marginTop:'19px'
    },
    md: {
        marginTop:'9px'
    },
    lg: {
        marginTop:'3px'
    }
};