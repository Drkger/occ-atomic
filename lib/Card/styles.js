import colors from '../subatomic/colors';
import fonts from '../subatomic/fonts';
import shadows from '../subatomic/shadows';

const styles = {
    card: {
        background:colors.white,
        borderRadius:'5px',
        padding:'30px 20px'
    },
    cardNoPadding: {
        extend: 'card',
        padding:'0'
    }
};

const pushShadows = function() {
    for (let i = 0; i<7; i++) {
        styles[`shadow${i}`] = {
            boxShadow: shadows[`lvl${i}`]
        };
    }
};

pushShadows();

export default styles;