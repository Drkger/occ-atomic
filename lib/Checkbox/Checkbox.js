import React from 'react';
import PropTypes from 'prop-types';

import Text from '../Text';

class Checkbox extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            value: props.value
        };
        this.toggle = this.toggle.bind(this);
    }

    componentWillReceiveProps(props) {
        const { value } = props;
        if (value && this.state.value != value)
            this.setState({ value });
    }

    toggle() {
        const { onChange, id, disabled } = this.props;
        if (!disabled) {
            let value = !this.state.value;
            this.setState({ value });
            if (onChange) onChange(value, id);
        }
    }

    render() {
        const { value } = this.state;
        const { classes, label, right, disabled, id, className, style } = this.props;
        return (
            <div
                id={id}
                className={`${classes.cont}${value ? ` ${classes.active}` : ''}${disabled ? ` ${classes.disabled}` : ''}${className ? ` ${className}` : ''}`}
                onClick={this.toggle}
                style={style}>
                <div className={classes.check} />
                {label && <Text tag="label" className={classes.label}>{label}</Text>}
                {right && <Text tag="label" mid className={classes.right}>{right}</Text>}
            </div>
        );
    }
}

Checkbox.propTypes = {
    classes: PropTypes.object,
    /** Default value for the checkbox. */
    value: PropTypes.bool,
    /** The function to call when the value of the checkbox changes. */
    onChange: PropTypes.func,
    /** Disable the checkbox. */
    disabled: PropTypes.bool,
    /** Text to display next to the checkbox.  */
    label: PropTypes.node,
    /** Text to show on the right. */
    right: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number
    ]),
    id: PropTypes.string,
    className: PropTypes.string,
    style: PropTypes.object
};

export default Checkbox;
