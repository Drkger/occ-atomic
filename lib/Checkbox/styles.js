import colors from '../subatomic/colors';
import spacing from '../subatomic/spacing';
import icons from '../subatomic/icons';

export default {
    cont: {
        paddingTop:spacing.tiny,
        paddingBottom:spacing.tiny,
        boxSizing:'border-box',
        display:'flex',
        alignItems:'center',
        cursor:'pointer',
        outline:'0',
        '&:after': {
            content:'""',
            display:'table',
            clear:'both'
        },
        '&:hover $check:after': {
            background:icons.base(icons.check.icon([colors.grey200]))
        }
    },
    check: {
        width:spacing.base,
        height:spacing.base,
        position:'relative',
        '&:before': {
            content:'""',
            width:spacing.small,
            height:spacing.small,
            borderRadius:4,
            position:'absolute',
            top:'50%',
            left:'50%',
            transform:'translate(-50%, -50%)',
            border:`1px solid ${colors.grey200}`,
            background:colors.bgWhite
        },
        '&:after': {
            content:'""',
            width:18,
            height:18,
            marginTop:-1,
            position:'absolute',
            top:'50%',
            left:'50%',
            transform:'translate(-50%, -50%)',
            transition:'0.3s all',
            background:'transparent'
        }
    },
    active: {
        '& $check:after': {
            background:[icons.base(icons.check.icon([colors.prim])), '!important']
        }
    },
    disabled: {
        opacity:0.4,
        pointerEvents:'none'
    },
    label: {
        marginLeft:spacing.tiny,
        cursor:'pointer'
    }
};
