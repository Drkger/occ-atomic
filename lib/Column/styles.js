import grid from '../subatomic/grid';

const styles = {
    col: {
        boxSizing:'border-box',
        position:'relative',
        minHeight:'1px',
        paddingRight:'15px',
        paddingLeft:'15px'
    }
};

const pushGrid = function() {
    const sm = `@media (min-width: ${grid.sm}px)`;
    const md = `@media (min-width: ${grid.md}px)`;
    const lg = `@media (min-width: ${grid.lg}px)`;
    for (let i = 1; i <= 12; i++) {
        styles[`xs${i}`] = {
            float:'left',
            width:`${100/12*i}%`
        };
        styles[`offset-xs${i}`] = {
            marginLeft:`${100/12*i}%`
        };
        styles[`push-xs${i}`] = {
            left:`${100/12*i}%`
        };
        styles[`pull-xs${i}`] = {
            right:`${100/12*i}%`
        };
    };
    styles[sm] = {};
    styles[md] = {};
    styles[lg] = {};
    for(let i = 1; i <= 12; i++) {
        styles[sm][`sm${i}`] = {
            float:'left',
            width:`${100/12*i}%`
        };
        styles[sm][`offset-sm${i}`] = {
            marginLeft:`${100/12*i}%`
        };
        styles[sm][`push-sm${i}`] = {
            left:`${100/12*i}%`
        };
        styles[sm][`pull-sm${i}`] = {
            right:`${100/12*i}%`
        };
        styles[md][`md${i}`] = {
            float:'left',
            width:`${100/12*i}%`
        };
        styles[md][`offset-md${i}`] = {
            marginLeft:`${100/12*i}%`
        };
        styles[md][`push-md${i}`] = {
            left:`${100/12*i}%`
        };
        styles[md][`pull-md${i}`] = {
            right:`${100/12*i}%`
        };
        styles[lg][`lg${i}`] = {
            float:'left',
            width:`${100/12*i}%`
        };
        styles[lg][`offset-lg${i}`] = {
            marginLeft:`${100/12*i}%`
        };
        styles[lg][`push-lg${i}`] = {
            left:`${100/12*i}%`
        };
        styles[lg][`pull-lg${i}`] = {
            right:`${100/12*i}%`
        };
    }
};

pushGrid();

export default styles;