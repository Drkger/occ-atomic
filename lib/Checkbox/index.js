import injectSheet from 'react-jss';
import Checkbox from './Checkbox';
import styles from './styles';

export default injectSheet(styles)(Checkbox);
