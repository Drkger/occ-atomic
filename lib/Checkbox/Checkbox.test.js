import React from 'react';
import { shallow, mount } from 'enzyme';
import Checkbox from './Checkbox';
import styles from './styles';
import CheckboxJSS from './index';

const reduceClasses = (prev, curr) => Object.assign({}, prev, { [curr]: curr });

const classes = Object.keys(styles).reduce(reduceClasses, {});

describe("Checkbox", () => {

    it('matches the snapshot', () => {
        const wrapper = shallow(
            <Checkbox classes={classes} />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('has the right state', () => {
        const wrapper = shallow(
            <Checkbox classes={classes} />
        );
        expect(wrapper.prop('className')).toEqual('cont');
        wrapper.setState({value: true});
        expect(wrapper.prop('className')).toEqual('cont active');
    });

    it('appears indeterminated', () => {
        const wrapper = shallow(
            <Checkbox classes={classes} />
        );
        expect(wrapper.prop('className')).toEqual('cont');
        wrapper.setState({indeterminated: true});
        expect(wrapper.prop('className')).toEqual('cont indeterminated');
    });

    it('is disabled', () => {
        const wrapper = shallow(
            <Checkbox classes={classes} disabled />
        );
        expect(wrapper.prop('className')).toEqual('cont disabled');
    });

    it('toggles the value on click', () => {
        const wrapper = shallow(
            <Checkbox classes={classes} />
        );
        expect(wrapper.prop('className')).toEqual('cont');
        wrapper.simulate('click');
        expect(wrapper.prop('className')).toEqual('cont active');
    });

    it('calls onChange function', () => {
        const onChange = jest.fn();
        const wrapper = shallow(
            <Checkbox classes={classes} onChange={onChange} />
        );
        wrapper.simulate('click');
        expect(onChange.mock.calls.length).toBe(1);
    });

    it('shows the label', () => {
        const wrapper = mount(
            <Checkbox classes={classes} label="Label content" right="Right" />
        );
        expect(wrapper.find('.label').at(0).text()).toEqual('Label content');
    });

});

describe("CheckboxJSS", () => {

    it('matches the snapshot', () => {
        const wrapper = shallow(
            <CheckboxJSS />
        );
        expect(wrapper).toMatchSnapshot();
    });

});

describe("Checkbox styles", () => {

    it('matches the snapshot', () => {
        expect(styles).toMatchSnapshot();
    });

});
