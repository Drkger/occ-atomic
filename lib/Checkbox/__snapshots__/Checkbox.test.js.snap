// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Checkbox
    classes={
      Object {
        "active": "active",
        "check": "check",
        "cont": "cont",
        "disabled": "disabled",
        "label": "label",
        "overflow": "overflow",
        "right": "right",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="check"
        />,
        undefined,
        undefined,
      ],
      "className": "cont",
      "id": undefined,
      "onClick": [Function],
      "style": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "check",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      undefined,
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="check"
          />,
          undefined,
          undefined,
        ],
        "className": "cont",
        "id": undefined,
        "onClick": [Function],
        "style": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "check",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        undefined,
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Checkbox styles matches the snapshot 1`] = `
Object {
  "active": Object {
    "& $check:after": Object {
      "background": Array [
        "url(data:image/svg+xml;base64,PHN2ZyB2ZXJzaW9uPSIxLjEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgeG1sbnM6eGxpbms9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiIHg9IjBweCIgeT0iMHB4IgogICAgICAgICAgICAJIHZpZXdCb3g9IjAgMCAyNCAyNCIgc3R5bGU9ImVuYWJsZS1iYWNrZ3JvdW5kOm5ldyAwIDAgMjQgMjQ7IiB4bWw6c3BhY2U9InByZXNlcnZlIj4KICAgICAgICAgICAgPHBhdGggZmlsbD0iIzA4M2NhZSIgZD0iTTE4LjI4MDI1MTgsNy4zMDU3NjQ4IEwxOS43MTk3NDgyLDguNjk0MjM1MiBMMTEuMDM4NzcyNCwxNy42OTQyMzUyIEMxMC42NzM3NTY4LDE4LjA3MjY2NSAxMC4wNzgwMzIzLAogICAgICAgICAgICAxOC4xMDM3OTczIDkuNjc1NTU5NzIsMTcuNzY1NDc2IEw1LjM1NjUzNTU2LDE0LjEzNDg3NDMgTDYuNjQzNDY0NDQsMTIuNjAzOTIyMyBMMTAuMjQ3NjE2NywxNS42MzM1OTc3IEwxOC4yODAyNTE4LDcuMzA1NzY0OCBaIi8+CiAgICAgICAgICAgIDwvc3ZnPg==)",
        "!important",
      ],
    },
  },
  "check": Object {
    "&:after": Object {
      "background": "transparent",
      "content": "\\"\\"",
      "height": 18,
      "left": "50%",
      "marginTop": -1,
      "position": "absolute",
      "top": "50%",
      "transform": "translate(-50%, -50%)",
      "transition": "0.3s all",
      "width": 18,
    },
    "&:before": Object {
      "background": "#ffffff",
      "border": "1px solid #dddddd",
      "borderRadius": 4,
      "content": "\\"\\"",
      "height": 16,
      "left": "50%",
      "position": "absolute",
      "top": "50%",
      "transform": "translate(-50%, -50%)",
      "width": 16,
    },
    "height": 24,
    "position": "relative",
    "width": 24,
  },
  "cont": Object {
    "&:after": Object {
      "clear": "both",
      "content": "\\"\\"",
      "display": "table",
    },
    "&:hover $check:after": Object {
      "background": "url(data:image/svg+xml;base64,PHN2ZyB2ZXJzaW9uPSIxLjEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgeG1sbnM6eGxpbms9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiIHg9IjBweCIgeT0iMHB4IgogICAgICAgICAgICAJIHZpZXdCb3g9IjAgMCAyNCAyNCIgc3R5bGU9ImVuYWJsZS1iYWNrZ3JvdW5kOm5ldyAwIDAgMjQgMjQ7IiB4bWw6c3BhY2U9InByZXNlcnZlIj4KICAgICAgICAgICAgPHBhdGggZmlsbD0iI2RkZGRkZCIgZD0iTTE4LjI4MDI1MTgsNy4zMDU3NjQ4IEwxOS43MTk3NDgyLDguNjk0MjM1MiBMMTEuMDM4NzcyNCwxNy42OTQyMzUyIEMxMC42NzM3NTY4LDE4LjA3MjY2NSAxMC4wNzgwMzIzLAogICAgICAgICAgICAxOC4xMDM3OTczIDkuNjc1NTU5NzIsMTcuNzY1NDc2IEw1LjM1NjUzNTU2LDE0LjEzNDg3NDMgTDYuNjQzNDY0NDQsMTIuNjAzOTIyMyBMMTAuMjQ3NjE2NywxNS42MzM1OTc3IEwxOC4yODAyNTE4LDcuMzA1NzY0OCBaIi8+CiAgICAgICAgICAgIDwvc3ZnPg==)",
    },
    "alignItems": "start",
    "boxSizing": "border-box",
    "cursor": "pointer",
    "display": "flex",
    "outline": "0",
    "paddingBottom": 8,
    "paddingTop": 8,
  },
  "disabled": Object {
    "opacity": 0.4,
    "pointerEvents": "none",
  },
  "label": Object {
    "cursor": "pointer",
    "flex": "1",
    "float": "left",
    "marginLeft": 8,
  },
  "overflow": Object {
    "overflow": "hidden",
    "textOverflow": "ellipsis",
    "whiteSpace": "nowrap",
  },
  "right": Object {
    "float": "right",
    "marginLeft": 8,
  },
}
`;

exports[`CheckboxJSS matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Jss(Checkbox) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "active": "Checkbox-active-0-1-38",
        "check": "Checkbox-check-0-1-37",
        "cont": "Checkbox-cont-0-1-36",
        "disabled": "Checkbox-disabled-0-1-39",
        "label": "Checkbox-label-0-1-40",
        "overflow": "Checkbox-overflow-0-1-42",
        "right": "Checkbox-right-0-1-41",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "active": "Checkbox-active-0-1-38",
          "check": "Checkbox-check-0-1-37",
          "cont": "Checkbox-cont-0-1-36",
          "disabled": "Checkbox-disabled-0-1-39",
          "label": "Checkbox-label-0-1-40",
          "overflow": "Checkbox-overflow-0-1-42",
          "right": "Checkbox-right-0-1-41",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
